activerecord/lib/active_record/autosave_association.rb

```diff
@@ -365,7 +365,12 @@ def normalize_reflection_attribute(indexed_attribute, reflection, index, attribu
       # Is used as a before_save callback to check while saving a collection
       # association whether or not the parent was a new record before saving.
       def before_save_collection_association
-        @new_record_before_save = new_record?
+        # If we defined a callback that updates the object before we defined the association, then this ivar may end up being
```

Comment:
I think this change is more needed because callbacks can be reentrant and what matters is the state of the record before the first callback, no after each callback. I think it is safe the remove the comment given that is what the documentation tells and now we have a test to make that explicit.

Reply:
Makes sense! Removed the comment.